README.md

```diff
@@ -91,97 +103,151 @@ cloning over SSH may provide a better experience (which requires
     git clone git@github.com:apple/swift.git
     ./swift/utils/update-checkout --clone-with-ssh
 
-#### CMake
-[CMake](https://cmake.org) is the core infrastructure used to configure builds of 
-Swift and its companion projects; at least version 3.4.3 is required. 
+### Building Swift
 
-On macOS, you can download the [CMake Binary Distribution](https://cmake.org/download),
-bundled as an application, copy it to `/Applications`, and add the embedded
-command line tools to your `PATH`:
+The `build-script` is a high-level build automation script that supports basic
+options such as building a Swift-compatible LLDB, building the Swift Package
+Manager, building for iOS, running tests after builds, and more. It also
+supports presets, which you can define for common combinations of build options.
 
-    export PATH=/Applications/CMake.app/Contents/bin:$PATH
-    
-On Linux, if you have not already installed Swift's [development
-dependencies](#linux-dependencies), you can download and install the CMake
-package separately using the following command:
+View the inline help to find out more, in particular the section "Typical uses"
 
-    sudo apt-get install cmake
-    
+    utils/build-script -h
 
-#### Ninja
-[Ninja](https://ninja-build.org) is the current recommended build system
-for building Swift and is the default configuration generated by CMake. [Pre-built
-packages](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages)
-are available for macOS and Linux distributions. You can also clone Ninja
-next to the other projects and it will be bootstrapped automatically:
+A basic command to build Swift with optimizations and run basic tests with
+Ninja:
 
-**Via HTTPS**
+    utils/build-script -r -t
 
-    git clone https://github.com/ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+### Development Environment
 
-**Via SSH**
+There are two primary build options to use with CMake, xcodebuild or Ninja.
+xcodebuild integrates nicely with Xcode.app, but Ninja is a bit faster and
+supports more environments.
 
-    git clone git@github.com:ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+To build using Xcode, run:
 
-#### Install via third-party packaging tool (macOS only)
+    utils/build-script -x
 
-**[Homebrew](https://brew.sh/)**
+The Xcode IDE can be used to edit the Swift source code, but it is not currently
+fully supported as a build environment for SDKs other than macOS. If you need to
+work with other SDKs, you'll need to create a second build using Ninja.
 
-    brew install cmake ninja
+To build using Ninja, run:
 
-**[MacPorts](https://macports.org)**
+    utils/build-script -R --debug-swift
```

Comment:
This will also build LLVM, Clang, and the Swift standard library with optimizations and asserts, but no debug symbols. Worth pointing out e.g. `utils/build-script -r --debug-swift-stdlib`, for people wanting to work on the standard library but want a optimized+asserts compiler. 

I think a better recommendation for beginners is `utils/build-script -r`, as a debug compiler or standard library is very slow, and I would expect many beginners initially to want something fast more than the ability to break in a debugger meaningfully. ^^^ This is complete opinion, feel free to ignore.